.circleci/config.yml

```diff
@@ -1,30 +1,65 @@
 # Main Config
 version: 2
+
+base_image: &base_image
+  image: circleci/node:8.11.3
+
+job_common: &job_common
+  docker:
+    - <<: *base_image
+
+save: &save
+  save_cache:
+    key: js-dependencies-{{ .Branch }}-{{ .Revision }}
+    paths:
+      - node_modules
+
+restore: &restore
+  restore_cache:
+    key: js-dependencies-{{ .Branch }}-{{ .Revision }}
+
 jobs:
+  lockfile_update:
+    <<: *job_common
+    steps:
+      - checkout
+      - <<: *restore
+      - greenkeeper-lockfile-update
+      - <<: *save
+
+  lockfile_upload:
+    <<: *job_common
+    steps:
+      - checkout
+      - <<: *restore
+      - greenkeeper-lockfile-upload
+      - <<: *save
+
   install:
-    docker:
-      - image: circleci/node:8.11.1
+    <<: *job_common
     steps:
       - checkout
-      - restore_cache:
-          keys:
-          - js-dependencies-{{ checksum "package.json" }}
-      - run: echo //registry.npmjs.org/:_authToken=${NPM_TOKEN} >> .npmrc
-      - run: npm install
-      - save_cache:
-          paths:
-            - node_modules
-          key: js-dependencies-{{ checksum "package.json" }}
+      - <<: *restore
+      - run: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" >> .npmrc
```

Comment:
You previously had `${NPM_TOKEN}`, does it need to be that?

Reply:
Quotes does the same thing as `${NPM_TOKEN}` I changed it because all the sample code in the circleCI docs